### PR TITLE
Handle errors on binding locations

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Handle errors in the binding locations, instead of the call handler.

#### Ticket Link
None

#### Related Pull Requests
None

#### Screenshots
None

### CODE REVIEW

File: actions/post_actions.jsx [hunk 1]
```diff
@@ -185,6 +185,7 @@ export function pinPost(postId) {
         if (rhsState === RHSStates.PIN) {
             addPostToSearchResults(postId, state, dispatch);
         }
+        return {data: true};
```
Comment (catalintomai) [inline actions/post_actions.jsx:188]: are these changes related to apps?

File: components/dot_menu/dot_menu.tsx [hunk 2]
```diff
@@ -290,6 +293,12 @@ export default class DotMenu extends React.PureComponent<Props, State> {
                 post_id: this.props.post.id,
                 root_id: this.props.post.root_id,
             },
+        }).then((res) => {
+            const callResp = (res as {data: AppCallResponse}).data;
```
Comment (catalintomai) [inline components/dot_menu/dot_menu.tsx:297]: can this snippet be factored out and reused throughout somehow?

File: actions/post_actions.jsx (hunk 1 above)
Comment (larkox) [reply actions/post_actions.jsx:188]: In order to be recognized in Typescript as Action func, they cannot return void, they must return either {data: any} or {error: any}.

File: components/dot_menu/dot_menu.tsx (hunk 2 above)
Comment (larkox) [reply components/dot_menu/dot_menu.tsx]: I prefer not. Each binding should handle its own errors, and I worry the moment these two bindings start diverging we need to refactor this in one more time.

File: components/post_view/embedded_bindings/select_binding/select_binding.tsx [hunk 3]
```diff
@@ -80,7 +84,31 @@ export default class SelectBinding extends React.PureComponent<Props, State> {
...
+        const callResp = (res as {data: AppCallResponse}).data;
+        const ephemeral = (message: string) => sendEphemeralPost(message, this.props.post.channel_id, this.props.post.root_id);
+        switch (callResp.type) {
+        case AppCallResponseTypes.OK:
+            if (callResp.markdown) {
+                ephemeral(callResp.markdown);
+            }
+            break;
+        case AppCallResponseTypes.ERROR: {
+            const errorMessage = callResp.error || this.props.intl.formatMessage({id: 'apps.error.unknown', defaultMessage: 'Unknown error happenned'});
```
Comment (catalintomai) [inline components/post_view/embedded_bindings/select_binding/select_binding.tsx:97]: curious, in what circumstances can we have `AppCallResponseTypes.ERROR` but `callResp.error` not set?

**Review by catalintomai — APPROVED** [review]

File: components/post_view/embedded_bindings/select_binding/select_binding.tsx (hunk 3 above)
Comment (larkox) [reply components/post_view/embedded_bindings/select_binding/select_binding.tsx:97]: In theory, only at the doAppCall level, in case the app returns a malformed response (error type with no error text).
I still make the check everywhere else to be extra cautious. Potentially, we can assume that error will be always set at this point, and move forward without this logic.

**Review by mickmister — APPROVED** [review]
LGTM! Awesome job with the intl :+1:
